### PR TITLE
AKS-23 Nodepool subnet size needs to accommodate maximum auto-scale settings 

### PR DESCRIPTION
**Pull request by dminkovski**
# Overview/Summary

This Pull Request introduces an Azure Resource Graph (ARG) query that identifies AKS node pools with auto-scaling enabled and checks if the associated subnet's size can accommodate the configured max-nodes setting. This enhancement helps in proactively identifying misconfigurations that could prevent AKS clusters from scaling as intended.

## Related Issues/Work Items

- APRL - AKS-23: Nodepool subnet size needs to accommodate maximum auto-scale settings 
- 29853 - Apps - AKS - Subnet sized for autoscale max

## This PR adds

1. A query to check if the subnet size is sufficient for the auto-scaling max-nodes setting in AKS node pools.
2. Improves visibility into potential AKS scaling issues due to subnet size limitations.
3. Facilitates proactive network planning and configuration for AKS clusters.

### Breaking Changes

1. None expected. The query addition is non-intrusive and does not modify existing infrastructure or settings.

### CODE REVIEW

**Comment by Dost2010** [comment]
Hello @dminkovski ,
Thank you for contributing. Can you please id column in  ARG query. For mor information please refer to [guide here](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing/#recommendation-categories)

File: docs/content/services/container/aks/code/aks-23/aks-23.kql [hunk 1]
```diff
@@ -1 +1,25 @@
...
+    | mv-expand subnets
+    | project id = tostring(subnets.id), addressPrefix = tostring(subnets.properties['addressPrefix'])
+    | extend subnetmask = toint(substring(addressPrefix, indexof(addressPrefix, '/')+1, string_size(addressPrefix)))
+    | extend possibleMaxNodeCount = toint(exp2(32-subnetmask) - 5)
+) on $left.subnetId == $right.id
+| project-away id, subnetmask
+| where possibleMaxNodeCount <= maxNodes
+| extend param1 = strcat(nodePoolName, " autoscaler upper limit: ", maxNodes)
+| extend param2 = strcat("ip addresses on subnet: ", possibleMaxNodeCount)
+| project recommendationId="aks-23", id=clusterId, name=clusterName, param1, param2
```
Comment (ejhenry) [inline docs/content/services/container/aks/code/aks-23/aks-23.kql:25]:
```suggestion
| project recommendationId="aks-23", id=clusterId, name=clusterName, param1, param2

```

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @dminkovski! Please include a screenshot of the successful testing of the query in the PR. Please also update the summary table on the _index.md page for AKS to indicate an ARG query is available for AKS-23.

**Comment by dminkovski** [comment]
**Testing Setup:**
`$LOCATION="westeurope"`
`$RESOURCE_GROUP="aks-23-rg"`
1. `az network vnet create --name MyVnet --resource-group $RESOURCE_GROUP --address-prefix 10.0.0.0/16 --location $LOCATION`
2. `az network vnet subnet create --name MySmallSubnet --resource-group $RESOURCE_GROUP --vnet-name MyVnet --address-prefix 10.0.0.0/29`
3. `az aks create --resource-group $RESOURCE_GROUP --name MyAKSCluster --node-count 1 --enable-cluster-autoscaler --min-count 1 --max-count 10 --network-plugin kubenet --vnet-subnet-id /path/to/MySmallSubnet --service-cidr 10.0.1.0/24 --dns-service-ip 10.0.1.10 --pod-cidr 10.0.2.0/24`

**Test:**

image

File: docs/content/services/container/aks/code/aks-23/aks-23.kql
Comment (ejhenry) [inline docs/content/services/container/aks/code/aks-23/aks-23.kql:1]: @dminkovski please update the order of the ARG query column names to match the contribution guide. Our automation tooling is expecting the output to match the prescribed order. Thanks.

**Review by oZakari — APPROVED** [review]
LGTM

File: docs/content/services/container/aks/_index.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ The presented resiliency recommendations in this guidance include Aks and associ
...
 | [AKS-2 - Isolate system and application pods](#aks-2---isolate-system-and-application-pods)                                                                                                                                 |    Governance     |  High   | Preview |         Yes         |
 | [AKS-3 - Disable local accounts](#aks-3---disable-local-accounts)                                                                                                                         | Access & Security |  High   | Preview |         Yes         |
-| [AKS-4 - Configure Azure CNI networking for dynamic allocation of IPs](#aks-4---configure-azure-cni-networking-for-dynamic-allocation-of-ips)                                                                               |    Networking    | Medium  | Preview |         Yes         |
+| [AKS-4 - Configure Azure CNI networking for dynamic allo ation-of-ips)                                                                               |    Networking    | Medium  | Preview |         Yes         |
```
Comment (ejhenry) [inline docs/content/services/container/aks/_index.md:20]: @dminkovski Looks like this may have been an unintentional edit. Please update.

**Review by ejhenry — APPROVED** [review]
LGTM

**Pull request merged**